sockeye/decoder.py

```diff
@@ -91,18 +94,23 @@ def decode_step(self,
                     target: mx.sym.Symbol,
                     target_max_length: int,
                     source_encoded_max_length: int,
-                    *states: mx.sym.Symbol) \
-            -> Tuple[mx.sym.Symbol, mx.sym.Symbol, List[mx.sym.Symbol]]:
+                    *states: mx.sym.Symbol,
```

Comment:
What about changing the interface here to always return 
pre-output-layer symbol AND the logits?
The caller of this method can then decide whether it wants to use logit_inputs or logits or both. If one is not needed it gets out of scope and within the decoders returning both is no additional overhead. This would simplify the signature of this method a bit. 
It also would make implementations of decoders a bit less hard to read as you can get rid of all the `if then else None` lines.

Reply:
That's a fair point.  Right now, none of our decoders require extra computation to return both.